pandas/core/arrays/string_.py

```diff
@@ -327,6 +328,8 @@ def astype(self, dtype, copy=True):
             arr[mask] = "0"
             values = arr.astype(dtype.numpy_dtype)
             return FloatingArray(values, mask, copy=False)
+        elif isinstance(dtype, CategoricalDtype):
```

Comment:
i think the thing to do is catch ExtensionDtype here, then do

```
cls = dtype.construct_array_type()
return cls._from_sequence(self, dtype=dtype, copy=copy)
```

Reply:
You're right, I added another test for it